### DISCUSSION TITLE
Unreadable font for GitHub og image for Chinese (or CJK)

### DISCUSSION THREAD

**Discussion by NightFurySL2001** [community/community#60294] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

GitHub is a popular place for all people in the world attracting users from multiple languages to share the site. However, for the Chinese community, the og image generated is undesirable. The OG image generated when linking to any pages containing Chinese characters may cause the text to show up in a thin serif-like font (songti) that is unreadable in small sizes, especially rendered with the resolution given; sometimes text are instead rendered with a script font (kaiti) instead, which does not fit the sans-serif aesthetics. 

his is probably due to poorly written font stack in the preview HTML, where the chosen font does not support the missing character and leading all the text after it to fallback to an unknown font installed on the machine.

Suggestion: Please include or install [Source Han Sans](https://github.com/adobe-fonts/source-han-sans) on the machine running the OG preview image and in the rendering font stack. The font will cover most CJK usages with relative ease.

Below shows some images as shown at this time (2023/07/07).

![static og image](https://github.com/community/community/assets/33471049/110847ae-63c6-4c1c-acf6-48a53bed098d)

This is fetched from https://opengraph.githubassets.com/a37ccf7688ec5b3089d8244a35e231cfc3234e275543816500bbd22158326a10/lxgw/LxgwWenKai/releases/tag/v1.300 and the live view is as below:
![live og image](https://opengraph.githubassets.com/a37ccf7688ec5b3089d8244a35e231cfc3234e275543816500bbd22158326a10/lxgw/LxgwWenKai/releases/tag/v1.300)

---

![static og image](https://github.com/community/community/assets/33471049/c45afd8b-ff8f-4fbd-a3f5-9941d36c5fd5)

This is fetched from https://opengraph.githubassets.com/a37ccf7688ec5b3089d8244a35e231cfc3234e275543816500bbd22158326a10/buttaiwan/iansui/issues/12 and the live view is as below:
![live og image](https://opengraph.githubassets.com/a37ccf7688ec5b3089d8244a35e231cfc3234e275543816500bbd22158326a10/buttaiwan/iansui/issues/12)

For reference, this is a normal issue og image with sans-serif font (heiti)

![live og image normal](https://opengraph.githubassets.com/a37ccf7688ec5b3089d8244a35e231cfc3234e275543816500bbd22158326a10/buttaiwan/iansui/issues/14)

---

![static og image](https://github.com/community/community/assets/33471049/a2d60049-c98a-4d4e-b456-9f1df923a79e)

This demonstrated the font fallback issue quite nicely. This is fetched from https://opengraph.githubassets.com/a37ccf7688ec5b3089d8244a35e231cfc3234e275543816500bbd22158326a10/buttaiwan/iansui/releases/tag/v0.943 and the live view is as below:
![live og image](https://opengraph.githubassets.com/a37ccf7688ec5b3089d8244a35e231cfc3234e275543816500bbd22158326a10/buttaiwan/iansui/releases/tag/v0.943)

**Comment by Jiang10086** [comment]
Yes, the font in the preview image for CJK is bad.
